src/components/SeedInfo/SeedInfoViews/FungalShifts.tsx

```diff
@@ -1,83 +1,110 @@
+	const key = props.direction ? "to" : "from"
+
+	/*
+	Var 'props.materials' may contain multiple materials with the same display name.
+	This is because some materials like 'Flammable Gas' have static variants that are
+	pre-rendered in a scene. Static materials have their own material ID.
+
+	These materials should not be displayed twice.
+
+	To solve this, materials will be sorted in a new `Map` where the key is the material
+	name acts as the key and the material ids are stored in the value as an array.
+	*/
+
+	const materialsByName = new Map()
+	props.materials.forEach((matName, matId) => {
+		const ids = materialsByName.get(matName) ?? []
+		ids.push(matId)
+		materialsByName.set(matName, ids)
+	});
+
+	// The logic that returns a JSX component requires an `Array`
+	const materialsByNameArray = Array.from(materialsByName)
+
+	return(
+		<Stack>
+			{props.isFlask && <Flask />}
+			{materialsByNameArray.map(([matName, matIds]) => {
+				return(
+					<div key={`${matName}-a`}>
+						<div key={`${matName}-b`} className={matIds.some(props.isFavorite) ? 'text-info' : ''}>
+							{capitalize(matName)}
+							{' '}
+							{props.showId && `(${matIds.join(', ')})`}
+						</div>
+					</div>
+				)
+			})}
+		</Stack>
+	)
+}
+
 interface IShiftProps {
 	key: string | number;
 	data: ReturnType<FungalInfoProvider['provide']>[number];
 	shifted: boolean;
 	isFavorite: (id: string) => boolean;
 	setShifted: (shifted: boolean) => void;
+	materialProvider: MaterialInfoProvider;
 	getMaterial: (s: string) => any;
 }
 
 const Shift = (props: IShiftProps) => {
-	const { data, shifted, setShifted, getMaterial, isFavorite } = props;
-
+	const { data, shifted, setShifted, getMaterial, isFavorite, materialProvider } = props;
 	const [showId] = useContext(AlchemyConfigContext);
 
-	const From = (props: { materials: string[]; flaskFrom: boolean }) => {
-		const { materials } = props;
-
-		const mats = new Map();
-
-		for (const m of materials) {
-			const displayName = capitalize(getMaterial(m));
-			const arr = mats.get(displayName) || [];
-			arr.push(m);
-			mats.set(displayName, arr);
-		}
-
-		return (
-			<Stack>
-				{data.flaskFrom && <Flask />}
-				{[...mats.keys()].map(displayName => {
-					const ids = mats.get(displayName);
-					return (
-						<div
-							key={displayName}
-							className={classNames(ids.some(isFavorite) && 'text-info')}
-						>
-							{displayName} {showId && `(${ids.join(', ')})`}
-						</div>
-					);
-				})}
-			</Stack>
-		);
-	};
+	// TODO: More uniform if data.from and data.to is always an array?
+	const from: Array<string> = [data.from].flat()
+	const to: Array<string> = [data.to].flat()
+	// TODO: Maybe have getMaterial return already capitalized?
+	const fromMaterials = new Map(from.map(matId => { return [ matId, materialProvider.translate(matId) ]}));
+	const toMaterials = new Map(to.map(matId => { return [ matId, materialProvider.translate(matId) ]}));
```

Comment:
Passing in materialProvider is not needed, because the function getMaterial (which can b e renamed to getMaterialName or something like that) already does the exact same thing (see line 160)

Reply:
I passed materialProvider because later, when I also retrieve the color, I don't have to add a bunch of properties. I am more flexible as I can just use the methods of the provider, to get what I need.

Reply:
Resolved by rename

Reply:
Looking at the code again, I did completely replace this in the fungal-color branch. I updated this branch as well to make more clear what I was going for.